### PR TITLE
Merge psexec and psexec_psh

### PR DESCRIPTION
**Pull request by wvu**
1. Merged `psexec` and `psexec_psh` modules
2. Split up code into multiple methods
3. Added autodetection for PowerShell
4. Created new targets for the various vectors

### CODE REVIEW

**Comment by OJ** [comment]
I approve of this PR :+1:

File: modules/exploits/windows/smb/psexec.rb [hunk 1]
```diff
@@ -102,151 +109,217 @@ def exploit
...
+    handler
+    disconnect
+  end
+
+  # TODO: Move to lib?
+  def have_powershell?
+    if datastore['SHARE'] == 'ADMIN$'
+      path = 'System32\\WindowsPowerShell\\v1.0\\powershell.exe'
+    else
+      path = 'Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe'
```
Comment (wvu-r7) [inline modules/exploits/windows/smb/psexec.rb]: This is the assumed path. Should work for ```C$``` and the like. The location of ```powershell.exe``` is entirely dependent on the root of the share.

**Comment by wvu-r7** [comment]
I think this is ready to go.

**Comment by bcook-r7** [comment]
Looks good to me; thanks for working with me on all the testing iterations.

**Comment by wvu-r7** [comment]
And thank you for the merge!

**Comment by wvu-r7** [comment]
Additional fixes here: #6372.

**Comment by wvu-r7** [comment]
Follow-ups: #6427, #6742, #7274.

**Pull request merged**